### PR TITLE
Simplify driver

### PR DESCRIPTION
**Pull request by willmcgugan**
Simplifies the structure of the driver. The base class now doesn't require a Console, only a file.

This is because textual-serve will set a "virtual" driver of sorts, which doesn't have a Console per se.

Also added a bunch of docstrings.

### CODE REVIEW

File: src/textual/app.py [hunk 1]
```diff
@@ -507,12 +507,12 @@ def animate(
 
     @property
     def debug(self) -> bool:
-        """bool: Is debug mode is enabled?"""
+        """Is debug mode is enabled?"""
```
Comment (rodrigogiraoserrao) [inline src/textual/app.py:510]:
```suggestion
        """Is debug mode enabled?"""
```

File: src/textual/driver.py [hunk 2]
```diff
@@ -2,26 +2,33 @@
...
         size: tuple[int, int] | None = None,
     ) -> None:
-        self.console = console
+        """Initialize a driver.
+
+        Args:
+            file: A file-like object open for writing unicode.
+            target: The message target (expected to be the app).
+            debug: Enabled debug mode. Defaults to False.
+            size: Initial size of the terminal or None to detect. Defaults to None.
```
Comment (rodrigogiraoserrao) [inline src/textual/driver.py:29]:
```suggestion
            debug: Enabled debug mode.
            size: Initial size of the terminal or `None` to detect.
```

File: src/textual/drivers/headless_driver.py [hunk 3]
```diff
@@ -48,7 +59,8 @@ def send_size_event():
         send_size_event()
 
     def disable_input(self) -> None:
-        pass
+        """Start application mode."""
```
Comment (rodrigogiraoserrao) [inline src/textual/drivers/headless_driver.py:62]: I don't get it. The methdo `disable_input` also starts application mode..?

File: src/textual/drivers/linux_driver.py [hunk 4]
```diff
@@ -29,13 +26,22 @@ class LinuxDriver(Driver):
...
         size: tuple[int, int] | None = None,
     ) -> None:
-        super().__init__(console, target, debug=debug, size=size)
+        """Initialize a driver.
+
+        Args:
+            file: A file-like object open for writing unicode.
+            target: The message target (expected to be the app).
+            debug: Enabled debug mode. Defaults to False.
+            size: Initial size of the terminal or None to detect. Defaults to None.
```
Comment (rodrigogiraoserrao) [inline src/textual/drivers/linux_driver.py:41]:
```suggestion
            debug: Enabled debug mode.
            size: Initial size of the terminal or `None` to detect.
```

File: src/textual/drivers/linux_driver.py [hunk 5]
```diff
@@ -45,6 +51,11 @@ def __rich_repr__(self) -> rich.repr.Result:
         yield "debug", self._debug
 
     def _get_terminal_size(self) -> tuple[int, int]:
+        """Detect the terminal size
+
+        Returns:
+            The size of the terminal as a tuple of (WIDTH, HEIGHT)
```
Comment (rodrigogiraoserrao) [inline src/textual/drivers/linux_driver.py:57]:
```suggestion
        """Detect the terminal size.

        Returns:
            The size of the terminal as a tuple of (WIDTH, HEIGHT).
```

File: src/textual/drivers/linux_driver.py [hunk 6]
```diff
@@ -184,6 +208,7 @@ def disable_input(self) -> None:
             pass
 
     def stop_application_mode(self) -> None:
+        """Disable further input."""
```
Comment (rodrigogiraoserrao) [inline src/textual/drivers/linux_driver.py:211]:
```suggestion
        """Stop application mode, restore state."""
```

File: src/textual/drivers/windows_driver.py [hunk 7]
```diff
@@ -1,71 +1,76 @@
...
 
     def __init__(
         self,
-        console: "Console",
+        file: IO[str],
         target: "MessageTarget",
         *,
         debug: bool = False,
         size: tuple[int, int] | None = None,
     ) -> None:
```
Comment (rodrigogiraoserrao) [inline src/textual/drivers/windows_driver.py:23]: Docstring?

**Review by rodrigogiraoserrao — APPROVED** [review]

File: src/textual/drivers/headless_driver.py (hunk 3 above)
Comment (willmcgugan) [reply src/textual/drivers/headless_driver.py:62]: Cut n paste error

**Pull request merged**